dace/sdfg/state.py

```diff
@@ -730,7 +730,7 @@ class SDFGState(OrderedMultiDiConnectorGraph[nd.Node, mm.Memlet],
 
     location = DictProperty(
         key_type=str,
-        value_type=symbolic.pystr_to_symbolic,
```

Comment:
Changing back to symbolic doesn't directly work here because all value objects need to have the same type, which is not possible if one also wants to store the memorytype (since symbolic for the memorytype doesn't make much sense). I will change the code to use the 2 keys, but can we just keep string as type here?

Reply:
Just realized that I mixed something here: I'd like to keep string type for the locationproperty of arrays, not for state. The reason I changed the type of the state locationproperty to string was because it lead to serialization errors for some of the HBM-tests. (I guess not all the values that are stored in there are symbols; e.g. the "is_FPGA_Kernel" is a boolean, not a symbol)

Reply:
Anyway I will change it back here, since this definitely does not belong in this PR